### DISCUSSION TITLE
Can Yabai return window to zoom-fullscreen mode after returning from Mac Full Screen? ex. Youtube browser window / Full Screen IDE

### DISCUSSION THREAD

**Discussion by nparkhe83** [asmvik/yabai#2352] [Q&A]
Hi, 

After watching Youtube Videos or any other Full Screen item on a web page, the browser returns back to the original size even if it was in `zoom-fullscreen` mode before.

Is it possible to change this behaviour such that the window returns back to the zoom-fullscreen mode after the Mac Full Screen mode (Green button) is toggled? I have observed this not only in browsers but also in case of IDEs and other applications.

Thanks for your help in advance 🙏 . 
If a script can be made, just point me towards the API and I will take it forward. Does Yabai track state of the open applications?

**Comment by Bellavene** [comment]
Sadly, but no. There is no system in yabai, to store window states across system invoked states.
I recommend switching to `yabai -m window --toggle windowed-fullscreen` instead of native one. Then it will be as you asked for.
As for traffic window buttons, you can change their behavior with another software (bettertouchtool, e.g.)
